app/main.py

```diff
@@ -1,6 +1,36 @@
+import os
+import requests
+from dotenv import load_dotenv
+
+
+load_dotenv()
+
+API_KEY = os.getenv("API_KEY")
+params = {
+    "key": API_KEY,
+    "q": "Paris",
+    "agi": "no"
+}
+URL = "http://api.weatherapi.com/v1/current.json"
+
+
 def get_weather() -> None:
-    # write your code here
-    pass
+    res = requests.get(URL, params=params)
+    if res.status_code == 200:
+        data = res.json()
+        name, country = data["location"]["name"], data["location"]["country"]
+        current_data = data["current"]
+        last_updated = current_data["last_updated"]
+        temp = current_data["temp_c"]
+        text = current_data["condition"]["text"]
+        print(f"{name}/{country} ")
+        print(f"{last_updated} Weather: {temp} Celsius, ")
+        print(f"{text}")
+
+    else:
+        print("Failed to retrieve data.")
+        print(f"Status code: {res.status_code}")
+        print(f"Reason: {res.reason}")
```

Comment:
Why Use a try Block Here?
Network Errors: The requests.get call can fail due to network issues, timeouts, or invalid URLs. Wrapping it in a try block ensures that these errors are caught and handled gracefully rather than causing your program to crash.

HTTP Errors: If the API returns an error response (e.g., 4xx or 5xx status codes), using response.raise_for_status() will raise an exception. This exception is caught by the except block to provide a meaningful error message.

JSON Parsing Errors: The response.json() call can fail if the response is not valid JSON. Wrapping this in a try block allows you to catch ValueError and handle it appropriately.

Missing Keys: The code assumes that certain keys will always be present in the JSON response. If the API response format changes or if the data is missing, this can raise a KeyError, which you can catch and handle to avoid crashes.

Unforeseen Errors: Catching a generic Exception ensures that any other unexpected errors are also handled, providing robustness to your function.